### PR TITLE
Expands test coverage for core functionality

### PR DESCRIPTION
**Pull request by valentinyanakiev**
### Describe the background of your pull request

This pull request significantly enhances the robustness and reliability of the codebase by adding extensive unit tests across several core components. These tests ensure critical functionalities behave as expected under various conditions, including error handling and edge cases.

Specifically, it introduces new tests that:
- Verify error handling and partial result generation within the `_refine_summarize` utility, covering mid-stream failures, initial failures, and empty inputs.
- Validate the `BodyOfKnowledgeSummaryStep`'s logic for section grouping to optimize LLM interactions and its behavior regarding inline persistence of summaries versus appending them to context chunks.
- Confirm the `LangChainLLMAdapter` correctly handles `asyncio.TimeoutError` by raising immediately without retries.
- Improve test coverage for the website ingestion plugin's HTML parsing, ensuring effective removal of boilerplate elements (e.g., `aside`, `form`, `dialog`, `noscript`, cookie/consent banners) while preserving semantic content.
- Validate the cross-page boilerplate deduplication process, including the filtering of documents that become empty after this operation.
- Confirm the web crawler accurately records the final URL after encountering HTTP redirects.

This work helps to solidify the stability and correctness of these foundational elements.

### Additional context

This is part of an effort to improve the overall test suite, aligning with the `chore/add-tests` branch.

### Governance

- [ ] Documentation is added
- [x] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

## Summary by CodeRabbit

* **Tests**
  * Added coverage for pipeline summary refinement: returns last partial summary on mid-stream failures, propagates on first-round failure, and handles empty input.
  * New Body-of-Knowledge tests: section grouping, max-size clamping, and persistence fallback/inline-storage behaviors.
  * Verified LLM timeout handling: timeouts surface as a timeout error and do not trigger retries.
  * Extended website ingest tests: HTML cleanup (boilerplate/cookie dialogs), cross-page deduplication, redirect URL handling, and async ingest filtering.

### CODE REVIEW

File: tests/plugins/test_ingest_website.py [hunk 1]
```diff
@@ -166,6 +166,258 @@ def test_strips_scripts(self):
...
+            import asyncio
+            mock_engine.return_value.run = lambda *a, **kw: asyncio.coroutine(
+                lambda: MagicMock(success=True, errors=[])
+            )()
+            await plugin.handle(event)
+
+        # IngestEngine.run receives the document list — check it doesn't include empty docs
+        run_call = mock_engine.return_value.run
+        # The engine was called — verify it was constructed
+        assert mock_engine.called
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply tests/plugins/test_ingest_website.py:374]: Fixed in 1aa5435. The test now uses `AsyncMock` and asserts both the document count (`len(docs) == 4`) and that all documents have non-empty content via `await_args`.

File: tests/plugins/test_ingest_website.py [hunk 2]
```diff
@@ -166,6 +166,258 @@ def test_strips_scripts(self):
...
+            knowledge_store=MockKnowledgeStorePort(),
+        )
+        event = make_ingest_website()
+
+        with patch("plugins.ingest_website.plugin.crawl", return_value=pages), \
+             patch("plugins.ingest_website.plugin.IngestEngine") as mock_engine:
+            import asyncio
+            mock_engine.return_value.run = lambda *a, **kw: asyncio.coroutine(
+                lambda: MagicMock(success=True, errors=[])
+            )()
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply tests/plugins/test_ingest_website.py]: Fixed in 1aa5435. Replaced all `asyncio.coroutine(lambda: ...)()` patterns with `AsyncMock(return_value=...)` in both `test_ingest_website.py` (4 occurrences) and `test_ingest_space.py` (1 occurrence).

File: tests/plugins/test_ingest_website.py [hunk 3]
```diff
@@ -166,6 +166,258 @@ def test_strips_scripts(self):
...
+                text=_html_page("Home", "Home content.", ["https://example.com/old-page"]),
+            )
+
+        with self._patch_transport(handler):
+            results = await crawl("https://example.com", page_limit=10)
+
+        # All URLs should be from response.url, not the request URL
+        urls = [r["url"] for r in results]
+        assert all(url.startswith("https://example.com") for url in urls)
+        assert len(results) >= 1
```
[the remark this answers is not in the document]
Comment (valentinyanakiev) [reply tests/plugins/test_ingest_website.py]: Fixed in 1aa5435. The test now uses an actual 302 redirect (`/old-page` → `/new-page`) with `follow_redirects=True` on the mock transport, and asserts that `/new-page` appears in the recorded URLs while `/old-page` does not.
